File: .local/statusbar/bat-perc.py
#!/usr/bin/env python


def main():
    perc: str
    charging: str
    icon: str = "  "

    # get current battery level
    with open("/sys/class/power_supply/BAT0/capacity", "r") as f:
        perc = f.read().strip()

    # get charging status
    with open("/sys/class/power_supply/AC/online", "r") as f:
        charging = f.read().strip()

    # determine icon based on battery level
    n: int = int(perc)

    if charging == "1":
        icon = " "
    else:
        if n > 10 and n < 40:
            icon = "  "
        elif n > 39 and n < 60:
            icon = "  "
        elif n > 59 and n < 90:
            icon = "  "
        elif n > 89:
            icon = "  "

    # create string to output
    bat_str: str = f"{icon} {perc}%  "
    print(bat_str)


if __name__ == "__main__":
    main()

File: .local/statusbar/bat-perc.c
#include <stdio.h>
#include <string.h>

int main()
{
    int perc;
    char state[12];
    char *icon = "  ";

    // get current battery level
    FILE *pf = fopen("/sys/class/power_supply/BAT0/capacity", "r");
    fscanf(pf, "%d", &perc);
    fclose(pf);

    // get charging status
    FILE *sf = fopen("/sys/class/power_supply/BAT0/status", "r");
    fscanf(sf, "%s", state);
    fclose(sf);

    // determine icon based on battery level and status
    if (strcmp(state, "Unknown") == 0)
    {
        icon = " ";
    }
    else if (!strcmp(state, "Charging"))
    {
        icon = " ";
    }

    else
    {
        if (perc > 10 && perc < 40)
        {
            icon = "  ";
        }
        else if (perc > 39 && perc < 60)
        {
            icon = "  ";
        }
        else if (perc > 59 && perc < 90)
        {
            icon = "  ";
        }
        else if (perc > 89)
        {
            icon = "  ";
        }
    }

    // output the data
    printf("%s%d%%  ", icon, perc);

    return 0;
}
